neuralmonkey/dataset/lazy_dataset.py

```diff
@@ -139,9 +160,13 @@ def series_ids(self) -> Iterable[str]:
         return (list(self.series_paths_and_readers.keys())
                 + list(self.preprocess_series.keys()))
 
-    def add_series(self, name: str, series: Iterable[Any]) -> None:
+    def add_lazy_series(self, name: str) -> None:
+        Dataset.add_series(self, name, [])
+
+    def add_series(self, name: str, series: List) -> None:
         raise NotImplementedError(
-            "Lazy dataset does not support adding series.")
+            "Lazy dataset does not support adding series. "
+            "Use add_lazy_series to add an empty series.")
```

Comment:
proc tady misto toho erroru neni obsah 'add_lazy_series'? vidim, ze se to pouziva jen v preprocess a tam si stejne externe zjistujes, jestli to neni lazy_dataset, takze by ten nemelo prekvapit, ze to pridava prazdne series, ne?

Reply:
Prvně jsem to tak měl, ale Python nedovoluje podědit a přetížit metodu (tj musí mít stejnou signaturu) a tím pádem, by to mělo `series: List`, který by se nepoužilo. Nejde to ani třeba s defaultním `None`. Mypy a pylint si vždy budou stěžovat. Takže to bylo buď volat v tom `helpers.py` tohle:

```python
dataset.add_series(name, [])
``` 

ale to bych se nezbavil toho komentáře `TODO ugly HACK`, protože to je hrozně neintuitivní.

No a nebo druhá možnost byla udělat to v metodě toho lazy datasetu, tak jsem se rozhod pro to.

Reply:
Ale možná bych to ještě rethinkoval. nicméně nevim jestli to ten kód nějak výrazně zlepší.

Reply:
Nech to tak. Nelibi se mi to, ale taky me nenapada, jak to jinak resit.